libs/ui/design-system/src/lib/copy-wrapper.tsx

```diff
@@ -0,0 +1,112 @@
+'use client'
+
+import { CheckCircle, Copy } from 'lucide-react'
+import { type FC, type PropsWithChildren, useRef, useState } from 'react'
+import { toast } from 'sonner'
+
+type Position = 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left'
+
+interface CopyWrapperProps {
+  showCopyButton?: boolean
+  position?: Position
+  onCopySuccess?: () => void
+  onCopyError?: (error: Error | string) => void
+  className?: string
+  buttonClassName?: string
+}
+
+const getPositionStyles = (position: Position): string => {
+  const base = 'absolute p-2'
+  const positions = {
+    'top-right': `${base} top-2 right-2`,
+    'top-left': `${base} top-2 left-2`,
+    'bottom-right': `${base} bottom-2 right-2`,
+    'bottom-left': `${base} bottom-2 left-2`,
+  }
+  return positions[position]
+}
+
+export const CopyWrapper: FC<PropsWithChildren<CopyWrapperProps>> = ({
+  children,
+  showCopyButton = true,
```

Comment:
what's the purpose of this prop?
what other way do we get the user to copy stuff?

Reply:
It was just fast iteration, I was about to get rid of it. :)